plugins/modules/azure_rm_privatednszone_vnetlink.py

```diff
@@ -0,0 +1,322 @@
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The Virtual network link ID.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/
+                     Microsoft.Network/privateDnsZones/privatezone.com/virtualNetworkLinks/vnetlink1"
+        name:
+            description:
+                - Virtual network link name.
+            returned: always
+            type: str
+            sample: 'vnetlink1'
+        location:
+            description:
+                - The Azure Region where the resource lives.
+            returned: always
+            type: str
+            sample: global
+        etag:
+            description:
+                - The etag of the virtual network link.
+            returned: always
+            type: str
+            sample: 692c3e92-a618-46fc-aecd-8f888807cd6c
+        tags:
+            description:
+                - Resource tags.
+            returned: always
+            type: list
+            sample: [{"key1": "value1"}]
+        virtual_network:
+            description:
+                - Reference to virtual network.
+            returned: always
+            type: dict
+            sample: {
+                "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/
+                       providers/Microsoft.Network/virtualNetworks/MyAzureVNet"
+            }
+        registration_enabled:
+            description:
+                - The status of auto-registration of virtual machine records in the virtual network in private DNS zone.
+            returned: always
+            type: bool
+            sample: true
+        provisioning_state:
+            description:
+                - The provisioning state of the resource.
+            returned: always
+            type: str
+            sample: Succeeded
+        virtual_network_link_state:
+            description:
+                - The status of the virtual network link.
+            returned: always
+            type: str
+            sample: Completed
+'''
+
+from ansible.module_utils.basic import _load_params
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase, HAS_AZURE, \
+    format_resource_id
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.polling import LROPoller
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMVirtualNetworkLink(AzureRMModuleBase):
+
+    def __init__(self):
+
+        _load_params()
+        # define user inputs from playbook
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            zone_name=dict(type='str', required=True),
+            virtual_network=dict(type='str'),
```

Comment:
Argument 'virtual_network' in argument_spec is not required, but is documented as being required

Reply:
corrected the documentation. virtual_network is required when state:present (create/update), so added it in required_if part. Thank you.